presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -482,4 +502,29 @@ public QueryContext getQueryContext(QueryId queryId)
     {
         return queryContexts.getUnchecked(queryId);
     }
+
+    public static class ExchangeClientManager
+    {
+        private final ExchangeClientSupplier supplier;
+        private final List<ExchangeClient> exchangeClients;
+
+        @VisibleForTesting
+        public ExchangeClientManager(ExchangeClientSupplier supplier)
```

Comment:
If only visible for testing, maybe we should make it package private?

Reply:
IIRC some tests are not in the same package :(